analyzer/src/funTest/assets/projects/synthetic/dotnet-direct-dependencies-only-expected-output.yml

```diff
@@ -16,40 +16,101 @@ project:
     path: "<REPLACE_PATH>"
   homepage_url: ""
   scopes:
-  - name: "net45"
-    dependencies:
-    - id: "NuGet::System.Threading.Tasks.Extensions:4.5.4"
-  - name: "netcoreapp3.1"
+  - name: "install"
     dependencies:
+    - id: "NuGet::Antlr:3.4.1.9004"
+    - id: "NuGet::Newtonsoft.Json:5.0.4"
     - id: "NuGet::System.Globalization:4.3.0"
     - id: "NuGet::System.Threading:4.0.11"
-  - name: "netcoreapp3.1-dev"
-    dependencies:
+    - id: "NuGet::System.Threading.Tasks.Extensions:4.5.4"
     - id: "NuGet::WebGrease:1.5.2"
+      dependencies:
+      - id: "NuGet::Antlr:3.4.1.9004"
+      - id: "NuGet::Newtonsoft.Json:5.0.4"
+    - id: "NuGet::foobar:1.2.3"
 packages:
+- id: "NuGet::Antlr:3.4.1.9004"
+  purl: "pkg:nuget/Antlr@3.4.1.9004"
+  authors:
+  - "Terence Parr <>"
+  declared_licenses:
+  - ""
+  declared_licenses_processed:
+    unmapped:
+    - ""
+  description: "ANother Tool for Language Recognition, is a language tool that provides\
+    \ a framework for constructing recognizers, interpreters, compilers, and translators\
+    \ from grammatical descriptions containing actions in a variety of target languages."
+  homepage_url: ""
+  binary_artifact:
+    url: ""
+    hash:
+      value: ""
+      algorithm: ""
+  source_artifact:
+    url: ""
+    hash:
+      value: ""
+      algorithm: ""
+  vcs:
+    type: ""
+    url: ""
+    revision: ""
+    path: ""
+  vcs_processed:
+    type: ""
+    url: ""
+    revision: ""
+    path: ""
+- id: "NuGet::Newtonsoft.Json:5.0.4"
+  purl: "pkg:nuget/Newtonsoft.Json@5.0.4"
+  authors:
+  - "James Newton-King <>"
+  declared_licenses:
+  - "LicenseUrl: http://json.codeplex.com/license"
```

Comment:
URLs to licenses should not be prefixed with `LicenseUrl: `. Always just take the declared license as-is, even if it's a URL.

If it's not an option to change this in `nuget-inspector` itself, the ORT integration should remove this prefix. We plan to address the issue of URLs being declared as licenses differently, see https://github.com/oss-review-toolkit/ort/pull/5259.

Reply:
Done!

Reply:
@sschuberth the data structure for licenses in the latest NuGet is much more complex: it used to be but is not longer simply a license URL. Instead, it can be a URL, or a license object with a type, with a possible expression and comes with a license type. We report the provided structure as-is with its field names because this is the only way that you can properly decode this and apply a correct license detection on top.
To get a better feel for the changes, see the license data structure at https://github.com/NuGet/NuGet.Client/blob/af6ad2a16c02e943993ab7e224473f4932d62c5b/src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/LicenseMetadata.cs
This is why we return something like this https://github.com/nexB/nuget-inspector/blob/main/src/nuget-inspector/Models.cs as a string which is YAML-formatted with values prefixed by their field.